Move Debug Logs From GitHub Gists to debuglogs.org

gasi opened the pull request:
In anticipation of GitHub’s deprecation of anonymous gists, we are moving our debug logs to https://debuglogs.org.

- [x] Publish debug logs to debuglogs.org.
- [x] Rename **Help > File a Bug** to **Report an Issue** for consistency with our debug log view as well as other apps such as Google Chrome.
- [x] Quickfix: Ensure `window.isFocused` always returns a boolean.

**Sample log:** https://debuglogs.org/0272bdd35288ed839ede88938a5290011150d42b56599efb5dd93ac80a2ac915

gasi commented on js/views/debug_log_view.js:
```diff
@@ -46,24 +46,24 @@
+      });
+      this.$('.loading').removeClass('loading');
+      view.render();
+      this.$('.link').focus().select();
```
Just like the old code, I didn’t add any extra error handling. It’d be nice to show a user friendly error message when submission fails but I’ll wait until we have a more declarative view system.

gasi commented on js/logging.js:
```diff
@@ -2,12 +2,10 @@
 const electron = require('electron');
 const bunyan = require('bunyan');
 const _ = require('lodash');
+const superagent = require('superagent');
```
The [`fetch` API is quite low-level](https://medium.com/@shahata/why-i-wont-be-using-fetch-api-in-my-apps-6900e6c6fe78) so I added `superagent` (by reputable authors) as a more high-level alternative that supports timeouts, form submission, and is also universal (isomorphic).

scottnonnenberg replied:
Hm. We may have to think a little bit more about whether `superagent` is the library we want to replace `fetch` with. One thing I do know is that, given that this will need to be ported to the Chrome app, we'll have to write it in jQuery either way.

gasi replied:
As discussed offline, I’ll change this to use `got`. As for Chrome, I will deal with that separately rather than having that app influence our code style / libraries for new changes.

scottnonnenberg commented on _locales/en/messages.json:
```diff
@@ -367,8 +367,8 @@
     "submitDebugLog": {
-      "message": "Submit debug log",
-      "description": "Menu item and header text for debug log modal, title case."
+      "message": "Debug log",
```
Where is this string used in the app? Can we remove it?

gasi replied:
It’s used as title of the debug log modal. The idea is to match the menu bar entry and the modal title for consistency but we currently use *Title Case* for menus and *Sentence case* for in-app titles, hence the two strings.

scottnonnenberg commented on js/focus_listener.js:
```diff
@@ -1,7 +1,7 @@
   'use strict';
 
-  var windowFocused;
+  var windowFocused = false;
```
Random changes! :0)

gasi replied:
Yes, but less randomization than opening a new PR 😉

gasi replied:
On a more serious note, I consider this part of continuous improvement based on the [broken window theory](https://blog.codinghorror.com/the-broken-window-theory/). But let me know if you prefer separate PRs for any orthogonal change. I was thinking we could allow smaller improvements within reason to reduce the overhead of separate branch creation, PR opening, CR, testing, etc.

gasi commented:
@scottnonnenberg @scottnonnenberg-signal: Thanks for the feedback. As discussed, I will re-implement the upload logic using `got` to make it more future-proof (promise support, custom CAs, etc.)

gasi commented:
Reimplemented upload logic using `got` but ran into an issue with the `got` & `form-data` integration where S3 responded with `411 Length Required` error. Will revisit it and try without `FormData` using `got` form support directly. Changes since last review: https://github.com/signalapp/Signal-Desktop/pull/2112/files/88eaf8603e8d4aa1402c84d63c780a2d9c0fe957..b0eb93de1afd06e44a27eb60ecc3302300690db7

gasi commented:
@scottnonnenberg: Revisited this and found the bug in `got`: https://github.com/sindresorhus/got/pull/466 Documented and stuck with our workaround of using `FormData::submit`. Ready for final review: https://github.com/signalapp/Signal-Desktop/pull/2112/files/b0eb93de1afd06e44a27eb60ecc3302300690db7..1ed52f721524d788ae0459d7bac9537131c6fd51

gasi commented:
Thanks for the review. Rebased on top of `debuglogs.org-autofix` and pushed before merging. No changes 👀

gasi commented:
Automatically closed by GitHub since target branch `debuglogs.org-autofix` was merged. Reopened against `development` as target branch.
